src/parser/SparqlParser.cpp

```diff
@@ -207,6 +210,54 @@ void SparqlParser::parseWhere(const string& str, ParsedQuery& query,
         continue;
       }
     } else if (inner[k] == '{') {
+      // look for a subquery
+      size_t h = k + 1;
+      while (std::isspace(inner[h])) {
+        h++;
+      }
+      if (ad_utility::getLowercase(inner.substr(h, 6)) == "select") {
+        size_t selectPos = h;
+        size_t endBracket = ad_utility::findClosingBracket(inner, k, '{', '}');
+        if (endBracket == std::numeric_limits<size_t>::max()) {
```

Comment:
I'd use `size_t(-1)` here to match the `return -1;` in `ad_utility::findClosingBracket()`.

Reply:
Done